File: scripts/clean.py
#!/usr/bin/env python3

#This script cleans the development environment by deleting everything but the needed files and
#folders. If you need any file or directory not to be deleted, add it to the following list.
crucial_directories = [ "scripts", "www", "config.xml", "package.json", "README.md", "LICENSE", \
	".gitignore", ".git" ]

#This script also removes Cordova platforms and plugins.

#Don't generate pycache directory. This can be removed.
from sys import dont_write_bytecode as dont_write_bytecode
dont_write_bytecode = True

from os import listdir, remove
from os.path import isdir
from shutil import rmtree
from core import run_w_err_handling, cd_project_root

def main():
	if input("This action will delete files. Continue? (Y/n) ") not in [ "Y", "y" ]:
		print("Action cancelled.")
		return
	
	cd_project_root()
	listing = listdir(".")

	#Cordova cleanup
	run_w_err_handling(["cordova", "platform", "remove", "browser"], \
		"Error removing platform browser. Stopping . . .")

	run_w_err_handling(["cordova", "plugin", "remove", "cordova-plugin-file"], \
		"Error removing cordova-plugin-file. Stopping . . .")

	#File deletion
	for file in listing:
		if file not in crucial_directories:
			if isdir(file):
				rmtree(file)
			else:
				remove(file)

	print("Environment successfully cleaned!")

if __name__ == "__main__":
	main()

File: scripts/clean.js
#!/usr/bin/env node

//This script cleans the development environment by deleting everything but the needed files and
//folders. If you need any file or directory not to be deleted, add it to the following list.
let crucialDirectories = [ "scripts", "www", "config.xml", "package.json", "README.md", "LICENSE",
	".gitignore", ".git" ];

//This script also removes Cordova platforms and plugins.

function main() {
	const { cdProjectRoot, prompt } = require("./core");
	const { readFile, writeFile, readdir, rm } = require("fs");

	//User confirmation
	prompt("This action will delete files. Continue? (Y/n) ", function(answer) {
		if (answer === "Y" || answer === "y"){
			cdProjectRoot();

			//Cordova cleanup (plugin and platform removal). Parse the package.json file to list the
			//plugins and platforms to remove.
			readFile("package.json", "utf8", function(err, contents) {
				if (err) {
					console.error("Failed to load package.json");
					process.exit(1);
				} else {
					try {
						let parsedJSON = JSON.parse(contents);

						try {
							parsedJSON.devDependencies   = {};
							parsedJSON.cordova.plugins   = {};
							parsedJSON.cordova.platforms = [];
							contents = JSON.stringify(parsedJSON, null, "\t");
						} catch {
							console.error("Failed to modify the contents of package.json");
							process.exit(1);
						}

						writeFile("package.json", contents, "utf8", function(err) {
							if (err) {
								console.error("Failed to write package.json to disk");
								process.exit(1);
							}
						});
					} catch {
						console.error("Failed to parse package.json");
						process.exit(1);
					}
				}
			});

			//Delete extra files and directories (while package.json is simultaneously modified)
			readdir(".", function(err, files) {
				if (err) {
					console.error("Error listing the repository's root directory");
				} else {
					for (let i = 0; i < files.length; ++i) {
						if (crucialDirectories.indexOf(files[i]) === -1) { //If not in crucial list
							rm(files[i], { recursive: true }, function(err) {
								if (err) {
									console.error("Failed to remove file/directory " + files[i]);
									process.exit(1);
								}
							});
						}
					}
				}
			});
		} else {
			console.log("Action cancelled");
		}
	});
}

if (require.main === module) {
	main();
}